ipn/ipnlocal/local_test.go

```diff
@@ -2173,3 +2173,42 @@ func TestOnTailnetDefaultAutoUpdate(t *testing.T) {
 		})
 	}
 }
+
+func TestTCPHandlerForDst(t *testing.T) {
+	b := newTestBackend(t)
+	ip := netip.MustParseAddr("100.100.103.100")
+
+	tests := []struct {
+		dst         netip.AddrPort
```

Comment:
I'd make this type string and use strings literals below in each test case for readability, even if that means not using `magicDNSIP`. I think that'd make reading the test cases easier.